diff --git a/configure b/configure
old mode 100755
new mode 100644
index 356bbe26f..99fe14b45
--- a/configure
+++ b/configure
@@ -178,6 +178,7 @@ all_platforms="${all_platforms} x86_64-win64-vs14"
 all_platforms="${all_platforms} x86_64-win64-vs15"
 all_platforms="${all_platforms} x86_64-win64-vs16"
 all_platforms="${all_platforms} x86_64-win64-vs17"
+all_platforms="${all_platforms} x86_64-win64-vs17-clangcl"
 all_platforms="${all_platforms} generic-gnu"
 
 # all_targets is a list of all targets that can be configured
@@ -760,6 +761,12 @@ process_toolchain() {
              enabled werror && gen_vcproj_cmd="${gen_vcproj_cmd} --enable-werror"
              all_targets="${all_targets} solution"
              INLINE="__inline"
+    
+             if echo "$toolchain" | grep -qi "clangcl"; then
+                 log_echo "Detected clang-cl toolchain; enabling /arch:SSE4.1"
+                 check_add_cflags "/arch:SSE4.1"
+                 check_add_cflags "-msse4.1"
+             fi
         ;;
     esac
 
