diff --git a/build/make/configure.sh b/build/make/configure.sh
index 679f12df7..9b84048d2 100644
--- a/build/make/configure.sh
+++ b/build/make/configure.sh
@@ -344,27 +344,42 @@ EOF
 }
 
 check_cflags() {
- log check_cflags "$@"
- check_cc -Werror "$@" <<EOF
+    log check_cflags "$@"
+    case "$CC" in
+        *cl | *clang-cl)
+            args=$(echo "$args" | sed 's/-Werror//g')
+            check_cc /WX $args <<EOF
 int x;
 EOF
+            ;;
+        *)
+            check_cc -Werror "$@" <<EOF
+int x;
+EOF
+            ;;
+    esac
 }
 
 check_cxxflags() {
-  log check_cxxflags "$@"
+    log check_cxxflags "$@"
 
-  # Catch CFLAGS that trigger CXX warnings
-  case "$CXX" in
-    *c++-analyzer|*clang++|*g++*)
-      check_cxx -Werror "$@" <<EOF
+    case "$CXX" in
+        *c++-analyzer|*clang++|*g++*)
+            check_cxx -Werror "$@" <<EOF
 int x;
 EOF
-      ;;
-    *)
-      check_cxx -Werror "$@" <<EOF
+            ;;
+        *cl | *clang-cl)
+            args=$(echo "$args" | sed 's/-Werror//g')
+            check_cxx /WX $args <<EOF
 int x;
 EOF
-      ;;
+            ;;
+        *)
+            check_cxx -Werror "$@" <<EOF
+int x;
+EOF
+            ;;
     esac
 }
 
