diff --git a/compat/windows/makedef b/compat/windows/makedef
index add8222d13..395a395f26 100755
--- a/compat/windows/makedef
+++ b/compat/windows/makedef
@@ -48,7 +48,7 @@ trap 'rm -f -- $libname' EXIT
 if [ -n "$AR" ]; then
     $AR rcs ${libname} $@ >/dev/null
 else
-    lib.exe -out:${libname} $@ >/dev/null
+    llvm-lib.exe -out:${libname} $@ >/dev/null
 fi
 if [ $? != 0 ]; then
     echo "Could not create temporary library." >&2
diff --git a/configure b/configure
old mode 100755
new mode 100644
index 7828381b5d..5ec4245a7f
--- a/configure
+++ b/configure
@@ -4767,15 +4767,15 @@ case "$toolchain" in
     msvc)
         cl_major_ver=$(cl.exe 2>&1 | sed -n 's/.*Version \([[:digit:]]\{1,\}\)\..*/\1/p')
         if [ -z "$cl_major_ver" ] || [ $cl_major_ver -ge 18 ]; then
-            cc_default="cl.exe"
-            cxx_default="cl.exe"
+            cc_default="clang-cl.exe"
+            cxx_default="clang-cl.exe"
         else
             die "Unsupported MSVC version (2013 or newer required)"
         fi
         ld_default="$source_path/compat/windows/mslink"
         windres_default="$source_path/compat/windows/mswindres"
         nm_default="dumpbin.exe -symbols"
-        ar_default="lib.exe"
+        ar_default="llvm-lib.exe"
         case "${arch:-$arch_default}" in
         aarch64|arm64)
             as_default="armasm64.exe"
@@ -5008,7 +5008,7 @@ msvc_common_flags(){
             -lstdc++)             ;;
             -l*)                  echo ${flag#-l}.lib ;;
             -LARGEADDRESSAWARE)   echo $flag ;;
-            -L*) [ "$_flags_type" = "link" ] && echo -libpath:${flag#-L} ;;
+            -L*) [ "$_flags_type" = "link" ] && echo "-libpath:${flag#-L}" ;;
             -Wl,*)                ;;
             *)                    echo $flag ;;
         esac
@@ -5262,11 +5262,11 @@ probe_cc(){
         _cflags_speed="-O2"
         _cflags_size="-O1"
         _cflags_noopt="-O1"
-        if $_cc -nologo- 2>&1 | grep -q Linker; then
+        if $_cc -nologo- 2>&1 | grep -q error; then
             _ld_o='-out:$@'
             _flags_filter=msvc_flags_link
         else
-            _ld_o='-Fe$@'
+            _ld_o='"-Fe$@"'
             _flags_filter=msvc_flags
         fi
         _cc_o='-Fo$@'
@@ -5276,6 +5276,7 @@ probe_cc(){
         _flags='-nologo'
         _cxxflags='-Zc:__cplusplus -EHsc'
         disable stripping
+        disable inline_asm_nonlocal_labels
     elif $_cc --version 2>/dev/null | grep -q ^cparser; then
         _type=cparser
         _ident=$($_cc --version | head -n1)
@@ -5366,6 +5367,9 @@ fi
 if VSLANG=1033 $ar 2>&1 | grep -q ^Microsoft; then
     arflags="-nologo"
     ar_o='-out:$@'
+elif $ar 2>&1 | grep -q "/llvmlibempty"; then
+    arflags="-nologo"
+    ar_o='-out:$@'
 elif $ar 2>&1 | grep -q "\[D\] "; then
     arflags="rcD"
     ar_o='$@'
@@ -6028,8 +6032,8 @@ case $target_os in
         SLIBSUF=".dll"
         SLIBNAME_WITH_VERSION='$(SLIBPREF)$(FULLNAME)-$(LIBVERSION)$(SLIBSUF)'
         SLIBNAME_WITH_MAJOR='$(SLIBPREF)$(FULLNAME)-$(LIBMAJOR)$(SLIBSUF)'
-        if test_cmd lib.exe -list; then
-            SLIB_EXTRA_CMD=-'lib.exe -nologo -machine:$(LIBTARGET) -def:$$(@:$(SLIBSUF)=.def) -out:$(SUBDIR)$(SLIBNAME:$(SLIBSUF)=.lib)'
+        if test_cmd llvm-lib.exe -list; then
+            SLIB_EXTRA_CMD=-'llvm-lib.exe -nologo -machine:$(LIBTARGET) -def:$$(@:$(SLIBSUF)=.def) -out:$(SUBDIR)$(SLIBNAME:$(SLIBSUF)=.lib)'
             if enabled x86_64; then
                 LIBTARGET=x64
             fi
@@ -7278,7 +7282,7 @@ enabled libvpx            && {
         die "libvpx enabled but no supported decoders found"
     fi
 }
-
+disable inline_asm_nonlocal_labels
 enabled libvvenc          && require_pkg_config libvvenc "libvvenc >= 1.6.1" "vvenc/vvenc.h" vvenc_get_version
 enabled libwebp           && {
     enabled libwebp_encoder      && require_pkg_config libwebp "libwebp >= 0.2.0" webp/encode.h WebPGetEncoderVersion
@@ -7788,6 +7792,7 @@ if [ -n "$lto" ]; then
     test "$cc_type" != "$ld_type" && die "LTO requires same compiler and linker"
     check_cflags  $lto
     check_ldflags $lto $cpuflags
+    disable inline_asm_nonlocal_labels
     disable inline_asm_direct_symbol_refs
     if test "$cc_type" = "clang"; then
         # Clang's LTO fails on Windows, when there are references outside
@@ -8462,6 +8467,7 @@ NOREDZONE_FLAGS=$noredzone_flags
 LIBFUZZER_PATH=$libfuzzer_path
 IGNORE_TESTS=$ignore_tests
 VERSION_TRACKING=$version_tracking
+LDEXEFLAGS += -wholearchive:avcodec.lib
 EOF
 
 map 'eval echo "${v}_FFLIBS=\$${v}_deps" >> ffbuild/config.mak' $LIBRARY_LIST
